File: python/3304-find-the-k-th-character-in-string-game-i.py
class Solution:
    def kthCharacter(self, k: int) -> str:
        length = 1
        while length < k:
            length *= 2
        ch = 'a'
        while length > 1:
            half = length // 2
            if k > half:
                k -= half
                ch = chr(((ord(ch) - ord('a') + 1) % 26) + ord('a'))
            length = half
        return ch

File: go/3304-find-the-k-th-character-in-string-game-i.go
func kthCharacter(k int) byte {
	length := 1
	ch := byte('a')
	for length < k {
		length *= 2
	}
	for length > 1 {
		half := length / 2
		if k > half {
			k -= half
			ch = (ch-'a'+1)%26 + 'a'
		}
		length = half
	}
	return ch
}